rotkehlchen/chain/evm/decoding/rainbow/decoder.py

```diff
@@ -0,0 +1,417 @@
+            events (list['EvmEvent']): A list of EVM events to be processed.
+            sender (ChecksumEvmAddress): The address of the sender involved in the events.
+        Returns:
+            in_event: Represents the receive the tokens from the swap
+            ou_event: Represents the spend the tokens to swap
+        """
+        in_event = out_event = None
+        for event in events:
+            if (
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == self.evm_inquirer.native_token
+            ):
+                # The "send eth" event is not enriched by enricher_rules,
+                # so it needs to be enriched here
+                self._enricher_event(event, sender)
+
+            if (
+                event.event_type == HistoryEventType.TRADE and
+                event.event_subtype == HistoryEventSubType.RECEIVE and
+                event.counterparty == CPT_RAINBOW_SWAPS
+            ):
+                in_event = event
+            elif (
+                event.event_type == HistoryEventType.TRADE and
+                event.event_subtype == HistoryEventSubType.SPEND and
+                event.counterparty == CPT_RAINBOW_SWAPS
+            ):
+                out_event = event
+
+        return in_event, out_event
+
+    def _create_fee_event(
+            self,
+            tx_hash: EVMTxHash,
+            timestamp: Timestamp,
+            fee_raw: int,
+            fee_asset: CryptoAsset,
+            sender: ChecksumEvmAddress,
+    ) -> 'EvmEvent':
+        """Create a new fee event for rainbow swap
+
+        Args:
+            tx_hash (EVMTxHash): transaction hash
+            timestamp (int): transaction timestamp
+            fee_raw (int): amount of fee charged in wei
+            fee_asset (Asset): fee asset
+            sender (ChecksumEvmAddress): sender address
+        """
+        return self.base.make_event(
+            tx_hash=tx_hash,
+            sequence_index=self.base.get_next_sequence_index(),
+            timestamp=timestamp,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=fee_asset,
+            amount=(fee_amount := asset_normalized_value(amount=fee_raw, asset=fee_asset)),
+            location_label=sender,
+            notes=f'Spend {fee_amount} {fee_asset.resolve_to_asset_with_symbol().symbol} as Rainbow Swap fees',  # noqa: E501
+            counterparty=CPT_RAINBOW_SWAPS,
+            address=self.router_address,
+        )
+
+    def _maybe_decode_swap_native_to_token(
+            self,
+            transaction: EvmTransaction,
+            event: 'EvmEvent',
+            decoded_events: list['EvmEvent'],
+            sender: ChecksumEvmAddress,
+            wrap_event: EvmTxReceiptLog,
+    ) -> None:
+        """
+        Decodes a swap transaction from native token to another token.
+        Check if the first log is deposit and the last is transfer log
```

Comment:
You didn't adjust this one. You don't seem to check on the transfer logs so it is wrong